src/lib/Toast/Toast.svelte

```diff
@@ -0,0 +1,154 @@
+<script lang="ts">
+  import { fly } from 'svelte/transition';
+
+  import { defaultToastProperties, type ToastProperties } from './properties';
+  import errorIcon from './error.svg';
+  import closeIcon from './close.svg';
+  import infoIcon from './info.svg';
+  import { createEventDispatcher, onDestroy, onMount } from 'svelte';
+
+  export let properties: ToastProperties = defaultToastProperties;
+  const dispatch = createEventDispatcher();
+
+  let visible = false;
+  let leftIcon = properties.leftIcon ?? null;
+  let timeout: ReturnType<typeof setInterval>;
+
+  properties = {
+    ...defaultToastProperties,
+    ...properties
+  };
+
+  async function hideToast() {
+    visible = false;
+  }
+
+  onMount(() => {
+    visible = true;
+    timeout = setInterval(hideToast, properties.duration);
+  });
+
+  onDestroy(() => {
+    clearInterval(timeout);
+  });
+
+  function getToastType(type: string): string | null {
```

Comment:
return enum toast Type  here instead of string

Reply:
updated.